web/yo/app/scripts/controllers/gene.js

```diff
@@ -3009,6 +2987,115 @@ angular.module('oncokbApp')
+                    var drugs = result;
+                    return drugs.filter(function (drug) {
+                        return (drug.drugName.toLowerCase().indexOf($query.toLowerCase()) != -1) || (drug.synonyms.join(',').toLowerCase().indexOf($query.toLowerCase()) != -1);
+                    })
+                })
+                .catch(
+                    function (error) {
+                        $scope.getDrugsError = true;
+                    }
+                )
+        }
+        $scope.addDruginTherapy = function(tag,index){
+            $scope.therapy[index].push(tag.drugName);
+            addTherValiEle(index);
+        }
+
+        $scope.removeDruginTherapy = function(tag,index){
+            var id = $scope.therapy[index].indexOf(tag.drugName);
+            if (id > -1) {
+                $scope.therapy[index].splice(id, 1);
+            }
+            addTherValiEle(index);
+        }
+
+        function addTherValiEle(index) {
+            if ($scope.therapy.length === index + 1 ){
+                $scope.therapy.push([]);
+            }
+            $scope.sameElement = false;
+            var therapyVali = [];
+            _.each($scope.therapy, function(element) {
+                element = element.sort();
+                element = element.join(' ').trim();
+                therapyVali.push(element);
+            });
+            for(var i=0;i<therapyVali.length;i++){
+                for(var j=i+1;j<=therapyVali.length;j++){
+                    if (therapyVali[i] == therapyVali[j]){
+                        $scope.sameElement = true;
+                    }
+                }
+            }
+        }
+
+        $scope.deleteTherapy = function(index){
+            if (index > 0){
+                $scope.therapy.splice(index, 1);
+            }
+        }
+
+        function isValidTreatment(indices, newTreatmentName) {
+            var isValid = true;
+            _.some(data.treatmentRef.treatments, function (treatment) {
+                if (treatment.name.toLowerCase() === newTreatmentName.toLowerCase()) {
+                    isValid = false;
+                    return true;
+                }
+            });
+            if (!isValid) {
+                $scope.sameTherapy = true;
+            }
+            return isValid;
+        }
+
+        $scope.save = function(){
+            while($scope.therapy[$scope.therapy.length-1]==''){
+                $scope.therapy.splice($scope.therapy.length-1, 1);
+            }
+            var therapyString = [];
+            var indices = data.indices;
+            if ($scope.sameElement === false){
+                _.each($scope.therapy,function(element){
+                    element = element.join(' + ').trim();
+                    therapyString.push(element);
+                });
+                var newTreatmentName = therapyString.join(', ').trim();
+                var treatment = new FirebaseModel.Treatment(newTreatmentName);
```

Comment:
@victoria34 could you pay attention to the below logic? Not familiar with that.

Reply:
@zhx828  The treatment is defined by Jiaojiao and it works well. Yi doesn't change this part of code.